source/common/http/http2/conn_pool.h

```diff
@@ -37,6 +38,26 @@ allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_
                  absl::optional<HttpServerPropertiesCache::Origin> origin = absl::nullopt,
                  Http::HttpServerPropertiesCacheSharedPtr http_server_properties_cache = nullptr);
 
+/**
+ * Abstract class for allocating reverse connection pools.
+ */
+class ReverseConnPoolFactory : public Config::UntypedFactory {
```

Comment:
Can we use `TypedFactory` and distinguish the extension by the proto message?

Reply:
Per the current state, it doesn't need any protobuf configuration. The configuration is handled separately through `ReverseConnectionListenerConfig`. Do you think we should add the blank proto and switch to `TypedFactory`?

I don't know the advantages of using one over the other. Will defer it to you :)